File: chat-service/src/main/java/org/example/chat/model/ChatMessage.java
package org.example.chat.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ChatMessage {
    
    public enum MessageType {
        ENTER, TALK, LEAVE
    }
    
    private MessageType type;
    private String roomId;
    private String sender;
    private String message;
    private long timestamp;
} 

File: proxy-kotlin/src/main/kotlin/com/chat/proxykotlin/domain/chat/ChatMessage.kt
package com.chat.proxykotlin.domain.chat

data class ChatMessage(
    val type: MessageType,
    val roomId: String,
    val userId: String?,
    val content: String,
    val timestamp: Long = System.currentTimeMillis()
)
enum class MessageType {
    CHAT,
    JOIN,
    LEAVE,
    SYSTEM
}